src/namepath.rs

```diff
@@ -14,6 +27,14 @@ impl<'src> Namepath<'src> {
       spaced: true,
     }
   }
+
+  pub fn push(&mut self, name: Name<'src>) {
+    self.0.push(name);
+  }
+
+  fn lexeme_iter(&self) -> impl ExactSizeIterator<Item = &str> {
```

Comment:
I don't think a helper function which is only used in one place adds much. `.all(|(a, b)| a.lexeme() == b.lexeme())` is fine.